src/inventory_ui.cpp

```diff
@@ -401,14 +454,329 @@ void inventory_selector::print_right_column() const
+    
+    //player &u = g->u;
+    
+    // First we want to print out worn stuff
+    for(size_t a = 0; a < worn.size(); a++) {
+        const itemstack_or_category &cur_entry = worn[a];
+        if(cur_entry.category == NULL) { // Blank object, skip over
+            continue;
+        }
+        if(cur_entry.it == NULL) { // This is a category header, print out category
+            if(had_first_cat) {
+                curr_line += 6; // Already printed first category, skip 5 lines
+                total_height += 6;
+            } else {
+                had_first_cat = true; // Printing first category, put right at top
+            }
+            const std::string name = cur_entry.category->name;
+            if(curr_line >= 3 && curr_line < TERMY) {
+                center_print(w_inv, curr_line, c_magenta, _("--- %s ---"), name.c_str()); // Center print header
+            }
+            curr_line++; // Move down a line
+            total_height++; // Track added height
+            row_i = 0; // Restart counting along row
+            cat_continued = false;
+            continue;
+        }
+        
+        const item &it = *cur_entry.it; // This is an item
+        
+        nc_color fill = (a == selected) ? selected_line_color : c_white;
+        nc_color invlet_color = (a == selected) ? selected_line_color :  // If selected, selected_line_color, otherwise (if hotkey assigned, yellow, otherwise white) 
+                (g->u.assigned_invlet.count(it.invlet) ? c_yellow : c_white);
+        nc_color name_color = (a == selected) ? selected_line_color : it.color_in_inventory();
+        
+        // Render the cell
+        if(curr_line >= 3 && curr_line < TERMY) {  // Top line
+        
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line, 1, c_white, cat_continued ? LINE_XXXO : LINE_OXXO);
+            } else {
+                mvwputch(w_inv, curr_line, 1 + (row_i * 25), c_white, cat_continued ? LINE_XXXX : LINE_OXXX);
+            }
+            if(!cat_continued) { // This is a continuation, no need to render top line
+                for(char x = 2; x < 26; x++) {
+                    mvwputch(w_inv, curr_line, x + (row_i * 25), c_white, LINE_OXOX);
+                }
+                mvwputch(w_inv, curr_line, 26 + (row_i * 25), c_white, LINE_OOXX);
+            } else { // Just edit the top right corner
+                mvwputch(w_inv, curr_line, 26 + (row_i * 25), c_white, LINE_XOXX);
+            }
+            // End Cell Border Stuff
+        }
+        
+        std::string item_name = it.label(); // Get object's display name
+        size_t count = 1;
+        if(cur_entry.slice != NULL) {
+            count = cur_entry.slice->size();
+            if(count > 1) {
+                item_name = it.label(count);
+            }
+        }
+        
+        if(curr_line >= 2 && curr_line + 1 < TERMY) {  // Second line
+            // Cell Border Stuff
```

Comment:
Instead of wrapping coding between "foo begin" and "foo end" comments, you could move it into a separate function (with a proper name) and call it from here.
```C++
void print_cell_border( ... ) {
    ...
}

...

{
    if(curr_line >= 2 && curr_line + 1 < TERMY) {
        print_cell_border( ... );
```

It does not need to be a free function, it could be a lambda.